Compiler warning when building Godot for the `server` platform

**Godot version:** Git https://github.com/godotengine/godot/commit/1fed266bf5452b30376db62495f4985f6975f2c1

**OS/device including version:** Fedora 30, Clang 8.0.0

**Issue description:** Clang reports a warning when compiling Godot for the `server` platform:

```text
In file included from platform/server/os_server.cpp:31:
platform/server/os_server.h:61:11: warning: private field 'ip_unix' is not used [-Wunused-private-field]
        IP_Unix *ip_unix;
```

I'm not sure if removing it entails anything, so I prefer making an issue first just to be sure.

**Steps to reproduce:** Build Godot for the `server` platform using `scons platform=server`.

--- a/platform/osx/os_osx.h
+++ b/platform/osx/os_osx.h
@@ -76,8 +76,6 @@ public:
 	List<String> args;
 	MainLoop *main_loop;
 
-	IP_Unix *ip_unix;
-
 #ifdef COREAUDIO_ENABLED
 	AudioDriverCoreAudio audio_driver;
 #endif

--- a/platform/server/os_server.h
+++ b/platform/server/os_server.h
@@ -58,7 +58,6 @@ class OS_Server : public OS_Unix {
 	bool grab;
 
 	virtual void delete_main_loop();
-	IP_Unix *ip_unix;
 
 	bool force_quit;
 

